Services/Mail/classes/AutoResponder/AutoResponder.php

```diff
@@ -0,0 +1,71 @@
+<?php declare(strict_types=1);
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+namespace ILIAS\Services\Mail\AutoResponder;
+
+use DateTimeImmutable;
+use DateInterval;
+
+class AutoResponder
+{
+    private int $sender_id;
+    private int $receiver_id;
+    private DateTimeImmutable $send_time;
+
+    public function __construct(int $sender_id, int $receiver_id, DateTimeImmutable $send_time)
```

Comment:
Would it be possible to make instances of this class immutable?

According to the docs:

> Repositories work well with immutable objects. Try to think about how data flows
through the program. A repository conceptually is a source and/or sink in some
data flow. A typical request might pull some entity from a repository, modify it
according to some user input and then put the modified version back to the
repository. If this request is modeled using immutable objects, the actual flow
of data can become more obvious then in typical stateful OOP implementations
since changes need to be propagated explicitly.